binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/main/java/org/springframework/cloud/stream/binder/reactorkafka/ReactorKafkaBinderHealthIndicator.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.reactorkafka;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.kafka.common.PartitionInfo;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.cloud.stream.binder.kafka.common.AbstractKafkaBinderHealthIndicator;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+
+/**
+ * {@link org.springframework.boot.actuate.health.HealthIndicator} for Reactor Kafka Binder.
+ *
+ * @author Soby Chacko
+ */
+public class ReactorKafkaBinderHealthIndicator extends AbstractKafkaBinderHealthIndicator {
+
+	private final ReactorKafkaBinder binder;
+
+	public ReactorKafkaBinderHealthIndicator(ReactorKafkaBinder binder,
+											ConsumerFactory<?, ?> consumerFactory) {
+		super(consumerFactory);
+		this.binder = binder;
+	}
+
+	@Override
+	protected ExecutorService createHealthBinderExecutorService() {
+		return Executors.newSingleThreadExecutor(
+			new CustomizableThreadFactory("reactor-kafka-binder-health-"));
+	}
+
+	protected Health buildTopicsHealth() {
+		try {
+			initMetadataConsumer();
+			Set<String> downMessages = new HashSet<>();
+			Set<String> checkedTopics = new HashSet<>();
+			final Map<String, ReactorKafkaBinder.TopicInformation> topicsInUse = ReactorKafkaBinderHealthIndicator.this.binder
```

Comment:
Ya, i was thinking the same in subsequent refactoring efforts. This is the first pass through the refactoring. We might carve out certain contracts in an interface and move it to core so that non-Kafka binders may also benefit.